cluster/kubernetes/resource/resource.go

```diff
@@ -70,7 +71,17 @@ func unmarshalObject(source string, bytes []byte) (resource.Resource, error) {
 	if err := yaml.Unmarshal(bytes, &base); err != nil {
 		return nil, err
 	}
+	r, err := unmarshalKind(base, bytes)
+	if err != nil {
+		return nil, &fluxerr.Error{
+			Type: fluxerr.User,
+			Err:  err,
+		}
```

Comment:
Needs help text I reckon. Something like 

> Could not parse "howdoes.yaml". This most likely means it is malformed YAML, possibly because you read the YAML spec and decided the universe has no inherent order or meaning.